### ISSUE TITLE
cpan2aur fails to create PKGBUILD for File::SWF

### ISSUE THREAD

**Issue by jnbek** [juster/perl-cpanplus-dist-arch#25]
When trying to build the PKGBUILD for:
https://metacpan.org/pod/SWF::File
Not sure how, but cpan2aur or CPANPLUS::Dist::Arch is adding a CRLF to the pkgbuild and refuses to continue.  
I get the following output:

```

([J:0]jjones@perlgeeks4u[/tmp]% cpan2aur SWF::File                                                                                                                                                                                                        [03/16/15][17:59:58])
==> Creating new source package for SWF::File...
  -> Looking up module for SWF::File on CPAN...
[MSG] Checking if source files are up to date
[MSG] Retrieving /home/jjones/.cpanplus/sourcefiles.s2.49_01.c0.9152.stored
[MSG] Using cached file '/home/jjones/.cpanplus/authors/id/Y/YS/YSAS/CHECKSUMS' on disk; ttl (3600s) is not exceeded
[MSG] Checksum matches for 'SWF-File-0.42.tar.gz'
# Extracting with ->_untar_bin
Running [/usr/bin/gzip -c -d -f /home/jjones/.cpanplus/authors/id/Y/YS/YSAS/SWF-File-0.42.tar.gz | /usr/bin/tar -tf -]...
SWF-File-0.42/
SWF-File-0.42/Changes
SWF-File-0.42/dumpswf.plx
SWF-File-0.42/lib/
SWF-File-0.42/lib/SWF/
SWF-File-0.42/lib/SWF/BinStream/
SWF-File-0.42/lib/SWF/BinStream/Codec/
SWF-File-0.42/lib/SWF/BinStream/Codec/Zlib.pm
SWF-File-0.42/lib/SWF/BinStream/Codec.pod
SWF-File-0.42/lib/SWF/BinStream/File.pm
SWF-File-0.42/lib/SWF/BinStream.pm
SWF-File-0.42/lib/SWF/Element.pm
SWF-File-0.42/lib/SWF/Element.pod
SWF-File-0.42/lib/SWF/File.pm
SWF-File-0.42/lib/SWF/Parser.pm
SWF-File-0.42/Makefile.PL
SWF-File-0.42/MANIFEST
SWF-File-0.42/META.yml
SWF-File-0.42/README
SWF-File-0.42/samples/
SWF-File-0.42/samples/img2swf.plx
SWF-File-0.42/samples/jpg2swf.plx
SWF-File-0.42/samples/linkext.plx
SWF-File-0.42/samples/readme.txt
SWF-File-0.42/samples/sample.plx
SWF-File-0.42/test.pl
Running [/usr/bin/gzip -c -d -f /home/jjones/.cpanplus/authors/id/Y/YS/YSAS/SWF-File-0.42.tar.gz | /usr/bin/tar -xf -]...
# Extraction succeeded
[MSG] DEFAULT 'filter_prereqs' HANDLER RETURNING 'sub return value'
==> ERROR: PKGBUILD contains CRLF characters and cannot be sourced.
[ERROR] makepkg returned abnormal status: 1
 at /usr/share/perl5/vendor_perl/CPANPLUS/Dist/Arch.pm line 463.
Failed to create CPANPLUS::Dist::Arch object at /usr/bin/vendor_perl/cpan2aur line 251.
([J:0]jjones@perlgeeks4u[/tmp]% 
                                                   
```

**Comment by juster** [comment]
That's pretty weird. I'll have to test it.

**Comment by juster** [comment]
P.S. Could you email me the PKGBUILD or add the PKGBUILD to the comments. If you comment it, mark the CRLFs with a command like this:

    tr '\r' '!' < PKGBUILD

**Comment by jnbek** [comment]
cpan2aur never actually creates the pkgbuild... :( it dies just before that point.

**Comment by jnbek** [comment]
The pkg in question is https://aur.archlinux.org/packages/perl-swf-file/ I just adopted it and was trying to update it, since it's using the build() instead of package(), etc... I'm pretty loyal to your utility ;-) don't really use anything else for perl-* packages. I wish there was some way to use Perl to manage all of the language pkgs, nodejs-, go-, etc :D

**Comment by juster** [comment]
The generated PKGBUILD should be somewhere under
~/.cpanplus/$perlver/build/perl-swf-file-$pkgver where $perlver is the
version of perl and $pkgver is the version of the SWF::File distribution.
I'll try to look at this more tomorrow, I'm still upgrading my virtual Arch
box.

On Mon, Mar 16, 2015 at 9:54 PM, John D Jones III <notifications@github.com>
wrote:

> The pkg in question is https://aur.archlinux.org/packages/perl-swf-file/
> I just adopted it and was trying to update it, since it's using the build()
> instead of package(), etc... I'm pretty loyal to your utility ;-) don't
> really use anything else for perl-* packages. I wish there was some way to
> use Perl to manage all of the language pkgs, nodejs-, go-, etc :D
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/juster/perl-cpanplus-dist-arch/issues/25#issuecomment-82029130>
> .
>



-- 
-Justin

**Comment by jnbek** [comment]
Oh alright, I'll have to grab that off my work system, probably tomorrow

**Comment by jnbek** [comment]
I tried to attach it, but Github wouldn't let me, so below is the PKGBUILD Contents according to Vim. Looks like it's in the pkgdesc field note the ^M: (vim's rendering of /r/n)

  1 # Contributor: John D Jones III <jnbek1972 -_AT_- g m a i l -_Dot_- com>
  2 # Generator  : CPANPLUS::Dist::Arch 1.29
  3
  4 pkgname='perl-swf-file'
  5 pkgver='0.42'
  6 pkgrel='1'
  7 pkgdesc="Create SWF file.^M"
  8 arch=('any')
  9 license=('PerlArtistic' 'GPL')
 10 options=('!emptydirs')
 11 depends=('perl-data-temporarybag>=0.08')
 12 makedepends=()
 13 url='http://search.cpan.org/dist/SWF-File'
 14 source=('http://search.cpan.org/CPAN/authors/id/Y/YS/YSAS/SWF-File-0.42.tar.gz')
 15 md5sums=('7319b35179edd24958fe1a92be8b3a0b')
 16 sha512sums=('3b4df24486654e86f885c7e81a97076e778a5e1a5afb9d6f37ecad4d22784b6e7d5cedc08b8e29dbdae4533e0c630c76a9ae5a43ced3f146e0ac392a4c68fa40')
 17 _distdir="SWF-File-0.42"
 18
 19 build() {
 20   ( export PERL_MM_USE_DEFAULT=1 PERL5LIB=""                 \
 21       PERL_AUTOINSTALL=--skipdeps                            \
 22       PERL_MM_OPT="INSTALLDIRS=vendor DESTDIR='$pkgdir'"     \
 23       PERL_MB_OPT="--installdirs vendor --destdir '$pkgdir'" \
 24       MODULEBUILDRC=/dev/null
 25
 26     cd "$srcdir/$_distdir"
 27     /usr/bin/perl Makefile.PL
 28     make
 29   )
 30 }
 31
 32 check() {
 33   cd "$srcdir/$_distdir"
 34   ( export PERL_MM_USE_DEFAULT=1 PERL5LIB=""
 35     make test
 36   )
 37 }
 38
 39 package() {
 40   cd "$srcdir/$_distdir"
 41   make install
 42
 43   find "$pkgdir" -name .packlist -o -name perllocal.pod -delete
 44 }
 45
 46 # Local Variables:
 47 # mode: shell-script
 48 # sh-basic-offset: 2
 49 # End:
 50 # vim:set ts=2 sw=2 et:

When you cat said file, it just munges things all to hell:

% cat PKGBUILD                                                                                                        [03/17/15][11:36:44])
# Contributor: John D Jones III <jnbek1972 -_AT_- g m a i l -_Dot_- com>
# Generator  : CPANPLUS::Dist::Arch 1.29

pkgname='perl-swf-file'
pkgver='0.42'
pkgrel='1'
"kgdesc="Create SWF file.
arch=('any')
license=('PerlArtistic' 'GPL')

Hope that's enough info for ya. I've seen it once or twice before, but the upstream released a new version later in the same day and the problem wasn't present.

**Comment by juster** [comment]
Thanks I see it's in the pkgdesc. makepkg's error was slightly innacurate. ^M is just a CR and there is no LF after it in this case.

I think that the pkgdesc is being scraped from a file with CRLF line endings. Probably from the META file. I will have to add a <code>$desc =~ s/\r//g</code> somewhere. I'll try to do it tonight.

**Comment by jnbek** [comment]
Awesome, thanks man. I default to ultra verbosity when doing bug reports, so forgive my pointing out the obvious :D

**Comment by juster** [comment]
Just released 1.30 with this fixed.
